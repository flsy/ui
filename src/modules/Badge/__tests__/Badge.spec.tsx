import { shallow } from 'enzyme';
import React from 'react';
import StyledBadge, { Badge } from '../Badge';

describe('<Badge />', () => {
  it('should render with styles', () => {
    const badge = shallow(<StyledBadge value={1} />);
    expect(badge).toMatchSnapshot();
  });
  it('should render correct value', () => {
    const badge = shallow(<Badge value={69} />);
    expect(badge.text()).toBe('69');
  });
  it('should render children', () => {
    const badge = shallow(
      <Badge value={69}>
        <div data-test-id="child">I am a child!</div>
      </Badge>,
    );
    expect(badge.find('[data-test-id="child"]').text()).toBe('I am a child!');
  });
});
