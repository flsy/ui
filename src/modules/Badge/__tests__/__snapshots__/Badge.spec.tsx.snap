// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render with styles 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

<Badge
  className="c0"
  value={1}
/>
`;
